redash/query_runner/elasticsearch.py

```diff
@@ -153,32 +138,85 @@ def parse_results(self, mappings, result_fields, raw_result, result_columns, res
 
         for h in raw_result["hits"]["hits"]:
             row = {}
-            for column in h["_source"]:
+
+            for field, column in ELASTICSEARCH_BUILTIN_FIELDS_MAPPING.iteritems():
+                if field in h:
+                    add_column_if_needed(mappings, field, column, result_columns, result_columns_index)
+                    row[column] = h[field]
+
+            column_name = "_source" if "_source" in h else "fields"
+            for column in h[column_name]:
                 if result_fields and column not in result_fields_index:
                     continue
 
-                if column not in result_columns_index:
-                    result_columns.append({
-                        "name" : column,
-                        "friendly_name" : column,
-                        "type" : mappings.get(column, "string")
-                    })
-                    result_columns_index[column] = result_columns[-1]
+                add_column_if_needed(mappings, column, column, result_columns, result_columns_index)
+
+                value = h[column_name][column]
+                row[column] = value[0] if isinstance(value, list) and len(value) == 1 else value
 
-                row[column] = h["_source"][column]
 
             if row and len(row) > 0:
                 result_rows.append(row)
 
-    def execute_simple_query(self, url, _from, mappings, result_fields, result_columns, result_rows):
+#
+#
+# Simple query example:
+#
+# - Query the index named "twitter"
+# - Filter by "user:kimchy"
+# - Return the fields: "@timestamp", "tweet" and "user"
+# - Return up to 15 results
+# - Sort by @timestamp ascending
+#
+# {
+#     "index" : "twitter",
+#     "query" : "user:kimchy",
+#     "fields" : ["@timestamp", "tweet", "user"],
+#     "size" : 15,
+#     "sort" : "@timestamp:asc"
+# }
+#
+#
+# Simple query on a logstash ElasticSearch instance:
+#
+# - Query the index named "logstash-2015.04.*" (in this case its all of April 2015)
+# - Filter by type:events AND eventName:UserUpgrade AND channel:selfserve
+# - Return fields: "@timestamp", "userId", "channel", "utm_source", "utm_medium", "utm_campaign", "utm_content"
+# - Return up to 250 results
+# - Sort by @timestamp ascending
+
+# {
+#     "index" : "logstash-2015.04.*",
+#     "query" : "type:events AND eventName:UserUpgrade AND channel:selfserve",
+#     "fields" : ["@timestamp", "userId", "channel", "utm_source", "utm_medium", "utm_campaign", "utm_content"],
+#     "size" : 250,
+#     "sort" : "@timestamp:asc"
+# }
+#
+#
```

Comment:
No need to keep this, I moved it to the docs: http://docs.redash.io/en/latest/usage/elasticsearch_querying.html